app/services/credit_card_resource/capture.rb

```diff
@@ -0,0 +1,52 @@
+class CreditCardResource
+  module Capture
+    def self.call(credit_card_resource, three_ds_challenge, atts) # rubocop:disable Metrics/MethodLength
+      auth_res = CreditCardResource::Authorize.(
+        credit_card_resource,
+        atts,
+        'capture',
+        three_ds_challenge
+      )
+
+      cap_atts = auth_res.slice(:customer_id, :transaction_id)
+      cap_res = Vantiv::Capture.call(get_txn_id(auth_res), cap_atts)
+
+      transaction_id = auth_res[:transaction_id]
+
+      VantivRequest::Create.capture(
+        transaction_id,
+        credit_card_resource,
+        auth_res[:order_id],
+        cap_res,
+        atts[:funds_transfer_id]
+      )
+
+      if cap_res['response'] == '000'
+        finalize_three_ds_challenge(three_ds_challenge, 'completed') if three_ds_challenge
+        auth_res[:credit_card_authorization].update(status: 'captured')
+        get_txn_id(auth_res)
+      else
+        CreditCardAuthorization::Reverse.call(
+          credit_card_authorization: auth_res[:credit_card_authorization],
+          auth_txn_id: auth_res[:transaction_id]
+        )
+        finalize_three_ds_challenge(three_ds_challenge, 'failed') if three_ds_challenge
+        error_code = ErrorCode::Build.(
+          'credit_card_resource.capture_error',
+          context: cap_res.data
+        )
+        Honeybadger.notify(error_code.message, context: error_code.context)
+        raise error_code
+      end
+    end
+
+    def self.get_txn_id(results)
+      results.dig(:cnp_txn_id) || results.dig(:txn_id) || results.dig(:litle_txn_id)
```

Comment:
`litle`? I've seen that a few times now. Did they misspell it in their api? Or does it stand for something else?

Also, .dig is unnecessary at one level.

```suggestion
      results[:cnp_txn_id] || results[:txn_id] || results[:litle_txn_id]
```

Reply:
I think it used to be what they were called at one time.  Old version is litle and new version is cnp